models/repo/license.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+func init() {
+	db.RegisterModel(new(RepoLicense))
+}
+
+type RepoLicense struct { //revive:disable-line:exported
+	ID          int64 `xorm:"pk autoincr"`
+	RepoID      int64 `xorm:"UNIQUE(s) INDEX NOT NULL"`
+	CommitID    string
+	License     string             `xorm:"VARCHAR(50) UNIQUE(s) INDEX NOT NULL"`
```

Comment:
I think it better to add a TODO for https://github.com/go-gitea/gitea/pull/24872/files#r1403957601, or It's hard for reader to understand your data struct designing.

```suggestion
  //  TODO: `INDEX` will be used to find repositories by license
	License     string             `xorm:"VARCHAR(50) UNIQUE(s) INDEX NOT NULL"`
```